app/web_ui/src/routes/(app)/docs/rag_configs/[project_id]/table_rag_config_row.svelte

```diff
@@ -0,0 +1,160 @@
+          border: "border-neutral/20",
+        }
+      }
+    }
+  }
+
+  $: status = ragProgressStore.get_status(rag_config.id || "")
+</script>
+
+{#if rag_progress && rag_config}
+  <tr class={row_hovered ? "hover" : ""}>
+    <!-- Step Info Card -->
+    <td class="align-top p-4 h-full">
+      <div class="flex flex-col gap-3">
+        <!-- Header -->
+        <div class="flex items-center justify-between">
+          <a
+            class="font-semibold text-lg text-base-content cursor-pointer link"
+            href={`/docs/rag_configs/${project_id}/${rag_config.id}/rag_config`}
+          >
+            {rag_config.name}
+          </a>
+        </div>
+
+        <!-- Description -->
+        <div class="space-y-1 text-xs text-base-content/70">
+          <div>
+            Extractor: {model_name(
+              rag_config.extractor_config?.model_name,
+              null,
+            ) || "N/A"} ({provider_name_from_id(
+              rag_config.extractor_config.model_provider_name,
+            ) || ""})
+          </div>
+          <div>
+            Chunk size: {rag_config.chunker_config.properties?.chunk_size ||
+              "N/A"}
+          </div>
+          <div>
+            Chunk overlap: {rag_config.chunker_config.properties
+              ?.chunk_overlap || "N/A"}
+          </div>
+          <div>
+            Embedding: {embedding_model_name(
+              rag_config.embedding_config.model_name,
+              rag_config.embedding_config.model_provider_name,
+            ) || "N/A"}
+          </div>
+          <div class="text-xs text-base-content/50">
+            Created {formatDate(rag_config.created_at)}
+          </div>
+        </div>
+      </div>
+    </td>
+
+    <!-- Progress Section -->
+    <td class="p-4 cursor-default align-top">
+      <div class="flex flex-col gap-3">
+        <!-- Overall Progress -->
+        <div class="flex items-center justify-between">
+          <div
+            class="badge {status_to_badge_props(status)
+              .color} badge-outline text-xs font-medium"
+          >
+            {#if status === "running"}
+              <div class="loading loading-spinner loading-xs mr-2"></div>
+            {/if}
+            <span>{status_to_badge_props(status).text}</span>
+          </div>
+          <span class="text-sm text-base-content/60">{completed_pct}%</span>
+        </div>
+        <progress
+          class="progress progress-primary bg-primary/20 w-full h-2"
+          value={rag_progress.total_document_completed_count || 0}
+          max={total_docs || 100}
+        ></progress>
+        {#if total_docs > 0}
+          <div class="text-xs text-base-content/50 text-start">
+            {rag_progress.total_document_completed_count || 0} of {total_docs} documents
+            processed
+          </div>
+        {/if}
+      </div>
+    </td>
+
+    <!-- Actions -->
+    <td class="p-4 cursor-default align-top">
+      <RunRagControl {rag_config} {project_id} />
```

Comment:
Move this into the status column? There's plenty of height, and it's very much related to status. Will feel more related, and will mean 1 less column which is good for small screens.

On second thought, the "RunRagControl" and badge are duplicative! Both display current status. When running we have 2 running spinners for the same thing, of 2 different colors. Let's make the RunRagControl the badge. Keep the nicer badge style, make it clickable, add a "see details" link affordance for complete state. Needs a big of design work but I think it cleans up a few issues.

Can we unify these two into one control?